analytics.js

```diff
@@ -894,6 +922,27 @@ class Analytics {
       // convert integrations object to server identified names, kind of hack now!
       transformToServerNames(rudderElement.message.integrations);
 
+      // Removing true values from clientSuppliedIntegrations except All
```

Comment:
Please add elaborated comments to explain the logic.

Reply:
updated